packages/react-server/core/components/RootContainer.js

```diff
@@ -23,9 +23,11 @@ RootContainer.defaultProps = {
 }
 
 RootContainer.flattenForRender = function(element) {
-	return [{containerOpen: getRootElementAttributes(element)}]
+	let attrs = getRootElementAttributes(element) || {};
+	attrs.tagName = attrs.tagName || 'div';
+	return [{containerOpen: true, attrs: attrs}]
```

Comment:
Something like `{containerOpen: tagName, attrs}` would be nicer here, to keep `attrs` and `tagName` separate.